io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java

```diff
@@ -101,15 +105,21 @@
      * @param endpointUrl is the URL to talk to.
      * @param restMaxSize initial (maximum) size of the rest call; if zero will send everything.
      * @param restMinSize minimum size for the call
-     * @param repositoryGroup the group to pass to the endpoint.
+     * @param repositoryGroup the group to pass to the endpoint, still can be used, but is replaced by
+     *        brewPullActive flag and mode when used in combination with DA 2.1 and above
+     * @param brewPullActive flag saying if brew pull should be used for version retrieval
+     * @param mode lookup mode, either STANDARD (default if empty) or SERVICE
```

Comment:
What happens if someone passes something else (e.g. `foobar`) to the flag? Should it be an enum / or have some other checks? Or is there something in DA we can call on to validate it?

Reply:
Also what does it do? Should we have test to validate it?

Reply:
The modes are dynamically configured in DA as needed.  Currently we plan to use 4 - PERSISTENT, TEMPORARY, SERVICE and SERVICE-TEMPORARY. It specifies allowed version suffixes, build categories and artifact qualities when looking up existing versions. If somebody passes a non-existing mode, it will fail in DA.

Reply:
So, inside PNC would repour pass the correct mode to pme, which passes it through to DA?

And the mode isn't hardcoded in DA, but configured on service start? Should pme handle non-existing mode specially ?

Reply:
Yes, Repour is responsible to pass the mode based on the build type etc. and the modes are configured in both DA and Repour at service start.
If a mode doesn't exist it should be returned by DA in an error response. If used directly by users they should be able to read it from the log (I expect the exception from the response should be printed in the terminal) and in case of running in PNC the sustaining team will find it in the logs to understand the problem and fix the config.
What kind of special handling would you suggest?

Reply:
None really :-) Let me rephrase, if an erroneous response is printed sufficiently clearly for people to understand, thats fine. I was just wondering if the error response should be parsed and in this case made really explicit what has gone wrong. But if you think its fine as is, thats fine.

Reply:
Actually when I tested it now it failed with `REST communication with http://da-master-devel.psi.redhat.com/da/rest/v-1 failed. Received response status -1 with message: Problem in HTTP communication with status code 500 and message Internal Server Error`. This is definitely suboptimal, but must be handled on DA side IMO.

Reply:
Sure, we can always do another iteration I guess if PME needs further changes

Reply:
I've added one more commit to handle this. It turned out the exception is in the response body, but PME did not print it even with debug on. Now it is printed like this:
```ERROR org.commonjava.maven.ext.cli.Cli - REST communication with http://da.example.com/da/rest/v-1 failed. Received response status 500 with message: javax.validation.ValidationException: Invalid mode name: a. Available modes are: [PERSISTENT, TEMPORARY]```